core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -176,10 +179,28 @@ class LogSegment private[log] (val log: FileRecords,
       throw new LogSegmentOffsetOverflowException(this, offset)
   }
 
+  /**
+   * Validate that offsets of records are monotonically increasing and are in [offsetLowerBound, offsetUpperBound] range.
```

Comment:
We are just verifying the offsets of record batches, not individual records.

Reply:
Right. Will fix the comment.